src/hexen/g_game.c

```diff
@@ -2049,7 +2057,7 @@ void G_InitNew(skill_t skill, int episode, int map)
     }
 
     // Set up a bunch of globals
-    if (!demoextend)
+    if ((!demoextend) || (!demorecording && !demoplayback))
```

Comment:
Where's the comment gone here? And still, `!demoextend` doesn't need a pair of extra parentheses just because you add some more conditions.

Reply:
>And still, !demoextend doesn't need a pair of extra parentheses just because you add some more conditions.

`if (!demoextend) || (!demorecording && !demoplayback)` gives me an error though :`expected expression before ‘||’ token`


Unless you mean something else?

Reply:
`if (!demoextend || (!demorecording && !demoplayback))`

Reply:
Ugh, sorry, I guess I'm overly tired right now.... thanks.